File: solutions/OreoFlavor/2026-08-07/medium/1209.java
import java.util.Scanner;
//1209
class Solution {
  static Scanner sc = new Scanner(System.in);
  public static void main(String[] args) {
    for(int tc = 0; tc < 10; tc++) {
      int testCase = sc.nextInt();
      
      int[] row = new int[100]; //행의 합
      int[] col = new int[100]; //열의 합
      int lr = 0; // 좌상단 -> 우하단 합
      int rl = 0; // 우상단 -> 좌하단 합

      int biggest = 0; //가장 큰 수

      int input = 0; //각 입력
      for(int i = 0; i < 100; i++) {
        for(int j = 0; j < 100; j++) {
          input = sc.nextInt();
          row[i] += input;
          col[j] += input;
          if(i == j) {
            lr += input;
          }
          if((i + j) == 99) {
            rl += input;
          }
        }
      }

      for(int i = 0; i < 100; i++) {
        if(row[i] > biggest) {
          biggest = row[i];
        }
        if(col[i] > biggest) {
          biggest = col[i];
        }
      }
      if(lr > biggest) {
        biggest = lr;
      }
      if(rl > biggest) {
        biggest = rl;
      }

      System.out.println("#" + testCase + " " + biggest);
    }
  }
}

File: solutions/HJK9810/2026-08-07/medium/1209.py
import sys
from pathlib import Path
input_path = Path(__file__).with_name("input.txt")
sys.stdin = open(input_path, "r", encoding="utf-8")

DEFAULT_SIZE = 100

def algorithm():
    boards = [list(map(int, input().split())) for _ in range(DEFAULT_SIZE)]

    row_sums = [sum(row) for row in boards]
    col_sums = [sum(col) for col in zip(*boards)]

    cross_sum = sum([boards[idx][idx] for idx in range(DEFAULT_SIZE)])
    reverse_cross_sum = sum([boards[DEFAULT_SIZE - idx - 1][DEFAULT_SIZE - idx - 1] for idx in range(DEFAULT_SIZE)])

    row_max = max(row_sums)
    col_max = max(col_sums)
    cross_max = max(cross_sum, reverse_cross_sum)

    return max(max(row_max, col_max), cross_max)

# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for _ in range(10):
    test_case = int(input())
    print(f"#{test_case} {algorithm()}")
